### ISSUE TITLE
Split builder module into multiple submodules

### ISSUE THREAD

**Issue by lfrancke** [stackabletech/operator-rs#317]
The current [builder.rs](https://github.com/stackabletech/operator-rs/blob/28a9015ea4c1e717bedc2fce918467461544b166/src/builder.rs) file is becoming very large.

We'd like to split it up into multiple modules.
I'm not sure if we want each builder in its own module or not but I'll leave that decision up to whoever implements this.

**Comment by razvan** [comment]
IMO this needs refinement. Is the size of the file really *the* problem ? Would it be better to split it into it's own crate, because it's largely independent of the framework (not used by it anywhere) ?

**Comment by maltesander** [comment]
Yeah i checked it is not used anywhere, so im in favor of extracting this code from the framework (which means we do not have to release the framework only for builder code changes). 

I did some quick research on what we have and tried to sort that a bit:

- Resource
  - ObjectMetaBuilder
  - OwnerReferenceBuilder
- Pod
  - PodBuilder
  - Volume
    - VolumeBuilder
    - VolumeMountBuilder
    - VolumeSource	
  - Container
    - ContainerBuilder
    - ContainerPortBuilder
  - Security
    - SecurityContextBuilder
    - PodSecurityContextBuilder
- ConfigMap
  - ConfigMapBuilder
- Event 
  - EventBuilder
- Node
  - NodeBuilder (not sure we want to keep this)


What we could extend (from checking the operators):
- Service
  - ServiceBuilder
  - ServicePortBuilder	
- StatefulSet
  - StatefulSetBuilder
- Daemonset
  - DaemonsetBuilder
- (Pod -> Volume ->) PersistantVolumeClaimBuilder
- (Pod -> Volume -> PersistantVolumeClaimBuilder) -> ResourceRequirements
- (Deployment -> DeploymentBuilder)

There are many more small structs which we could abstract builders from, not sure how granular we want to go.

Anyone has a better

**Comment by maltesander** [comment]
@teozkr anything against splitting this into its own crate?

**Comment by soenkeliebau** [comment]
@teozkr

**Comment by maltesander** [comment]
> As for splitting into another crate.. What would this help? You'd still need to do a release to use it in the operators, the only difference is that you'd be cutting a new `stackable-k8s-builder` (or whatever it ends up being named) release, rather than a `stackable-operator` release. Or for stuff that require changes in both, you'd now end up doing two different releases.

Currently, we do not use any builder stuff in the operator-rs and therefore could make it an extra crate. It probably helps with external contributors as well that are only interested in the builder and not the operator-rs itself?

On the other hand even if we only have the operators depend on it we have the whole k8s-openapi/kube-rs dependency hell again... which will not reduce the amount of tags we have to produce...

So yeah it probably makes sense for now to keep it in the framework?
